projects/lib/src/board/racingkingsboard.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+    This file is part of Cute Chess.
+
+    Cute Chess is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Cute Chess is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Cute Chess.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "racingkingsboard.h"
+#include "westernzobrist.h"
+
+namespace Chess {
+
+RacingKingsBoard::RacingKingsBoard()
+	: WesternBoard(new WesternZobrist())
+{
+}
+
+Board* RacingKingsBoard::copy() const
+{
+	return new RacingKingsBoard(*this);
+}
+
+QString RacingKingsBoard::variant() const
+{
+	return "racingkings";
+}
+
+QString RacingKingsBoard::defaultFenString() const
+{
+	return "8/8/8/8/8/8/krbnNBRK/qrbnNBRQ w - - 0 1";
+}
+
+bool RacingKingsBoard::vIsLegalMove(const Move& move)
+{
+	vMakeMove(move, nullptr);
+	Side opp = sideToMove().opposite();
+	bool givesCheck = inCheck(opp);
+	vUndoMove(move);
+
+	if (givesCheck)
+		return false;
+
+	return WesternBoard::vIsLegalMove(move);
+}
+
+/*! Returns true if the king of \a side is on the eighth rank */
+bool RacingKingsBoard::finished(Side side) const
+{
+	Square ksq = chessSquare(kingSquare(side));
+	return (ksq.rank() == 7);
+}
+
+/*! Returns true if the king of \a side can reach the eighth rank */
+bool RacingKingsBoard::canFinish(Side side)
+{
+	QVarLengthArray<Move> moves;
+	WesternBoard::generateMovesForPiece(moves, King, kingSquare(side));
+	for (const Move &m: moves)
```

Comment:
One coding convention nit-pick: we attach the reference/deference operator to the type; not the variable name. So the for loop should be `for (const Move& m : moves)`.